### PR TITLE
[multitenancy-manager] Reject deletion of a ProjectType if used in Projects

### PR DESCRIPTION
**Pull request by Dmitrykob**
## Description

Added a new verification webhook. It checks if the ProjectType is used in any project and sends an error when attempting to remove it.

## Why do we need it, and what problem does it solve?

  If it fixes an issue https://github.com/deckhouse/deckhouse/issues/5316

## What is the expected result?

When trying to delete a projectType used in a project, an error is displayed.

## Checklist
- [ ] The code is covered by unit tests.
- [ ] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: multitenancy-manager
type: fix
summary: When trying to delete a projectType used in a project, an error with project name displayed.
impact_level: default
```

### CODE REVIEW

File: ee/modules/160-multitenancy-manager/webhooks/validating/projectType [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    return 0
+  fi
+
+  cat <<EOF > "$VALIDATING_RESPONSE_PATH"
+{"allowed":true}
+EOF
+
+}
+
+hook::run "$@"
```
Comment (shvgn) [inline ee/modules/160-multitenancy-manager/webhooks/validating/projectType:53]: Do you really prefer bash over python?

Comment (Dmitrykob) [reply ee/modules/160-multitenancy-manager/webhooks/validating/projectType:53]: It's the same for me.
Moreover, the hook next to it is also written in bash.

**Review by RomanenkoDenys — COMMENTED** [review]
Please add correct Changelog entries section

**Review by RomanenkoDenys — APPROVED** [review]
LGTM

File: tools/go.mod [hunk 2]
```diff
@@ -15,3 +15,5 @@ replace (
 	github.com/deckhouse/deckhouse/dhctl => ../dhctl
 	github.com/deckhouse/deckhouse/go_lib/cloud-data => ../go_lib/cloud-data
 )
+
+replace github.com/flant/shell-operator => github.com/flant/shell-operator v1.3.2-0.20230911172604-d965e9a98cd7
```
Comment (yalosev) [inline tools/go.mod:19]: I'm confused with this one. Why do we need this replace?

Comment (Dmitrykob) [reply tools/go.mod]: It is not necessary. Leftovers from experimenting with versions shell-operators.
I'll be more careful in the future.

**Review by RomanenkoDenys — APPROVED** [review]
LGTM

**Review by yalosev — APPROVED** [review]

**Pull request merged**